lib/Sema/CodeSynthesis.cpp

```diff
@@ -763,17 +767,25 @@ void swift::addTrivialAccessorsToStorage(AbstractStorageDecl *storage,
 
   // Synthesize and type-check the body of the getter.
   synthesizeTrivialGetter(getter, storage, TC);
-  TC.typeCheckDecl(getter, true);
-  TC.typeCheckDecl(getter, false);
+  if (storage->getDeclContext()->isModuleScopeContext()) {
```

Comment:
I don't see why this is the right check. What's different about members? Why can't I just validate accessors at module scope?